Sources/ClientRuntime/Auth/HTTPAuthAPI/Signer.swift

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public protocol Signer {
+    func sign<IdentityT: Identity>(
```

Comment:
Generic in the method signature versus an associated type on the `Signer` protocol provides enough flexibility here?

Reply:
I believe so, yes.